### PR TITLE
[Azure] Application Insights Sampler

### PR DESCRIPTION
**Pull request by Austin-Tan**
Hi all, figuring this repo is the best place for this sampler. 

No current exporter publicly provides a sampler that is compatible with the Microsoft's Application Insights SDK, due to AI's use of the DJB2 Hash Algorithm. This sampler samples on just that, but should be used as a delegate to a parent sampler if decisions on parentcontext / parent sample status is desired.

Please let me know if you think there's a better location for this tool.
Thanks!

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/OpenTelemetry.Contrib.Exporter.ApplicationInsights.csproj [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
```
Comment (cijothomas) [inline src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/OpenTelemetry.Contrib.Exporter.ApplicationInsights.csproj:3]: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Contrib.Extensions.AWSXRay/OpenTelemetry.Contrib.Extensions.AWSXRay.csproj

Model this similar to the AWSExtension. This is *not* an exporter, but something like an extension. So it feel confusing to call it "Exporter"

File: test/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (reyang) [inline test/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests.csproj:4]: Do we target `net5.0` only or we want to cover more (e.g. `net461`)?

File: src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/ApplicationInsightsSampler.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Contrib.Exporter.ApplicationInsights
```
Comment (reyang) [inline src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/ApplicationInsightsSampler.cs:20]:
```suggestion
namespace OpenTelemetry.Contrib.Extensions.AzureMonitor
```

File: test/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests.csproj (hunk 2 above)
Comment (Austin-Tan) [reply test/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests/OpenTelemetry.Contrib.Exporter.ApplicationInsights.Tests.csproj]: Targeting 461 as well now

File: src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/ApplicationInsightsSampler.cs (hunk 3 above)
Comment (Austin-Tan) [reply src/OpenTelemetry.Contrib.Exporter.ApplicationInsights/ApplicationInsightsSampler.cs]: Refactored to this namespace

File: test/OpenTelemetry.Contrib.Extensions.AzureMonitor.Tests/OpenTelemetry.Contrib.Extensions.AzureMonitor.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.7.1" />
+    <PackageReference Include="MSTest.TestAdapter" Version="2.1.1" />
+    <PackageReference Include="MSTest.TestFramework" Version="2.1.1" />
+    <PackageReference Include="coverlet.collector" Version="1.3.0" />
+    <PackageReference Include="OpenTelemetry" Version="1.2.0-alpha1" />
```
Comment (reyang) [inline test/OpenTelemetry.Contrib.Extensions.AzureMonitor.Tests/OpenTelemetry.Contrib.Extensions.AzureMonitor.Tests.csproj:14]: Do we need "1.2.0-alpha1", I guess `[1.0.1,2.0.0)` would be what we want?

File: src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net461;net5.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry" Version="1.2.0-alpha1" />
```
Comment (reyang) [inline src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj:8]: Do we need "1.2.0-alpha1", I guess `[1.0.1,2.0.0)` would be what we want?

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by reyang** [comment]
@cijothomas I think this is ready to be merged. We probably don't need to release it till end of this year.

File: src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net461;net5.0</TargetFrameworks>
```
Comment (cijothomas) [inline src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj:4]: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Contrib.Extensions.AWSXRay/OpenTelemetry.Contrib.Extensions.AWSXRay.csproj#L5 need tagprefix to version correctly.

**Comment by cijothomas** [comment]
I'll merge now and do the minvertagprefix separately *before* actual release.

File: src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj (hunk 6 above)
Comment (cijothomas) [reply src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj:4]: change to netstandard20 as well.

Comment (cijothomas) [reply src/OpenTelemetry.Contrib.Extensions.AzureMonitor/OpenTelemetry.Contrib.Extensions.AzureMonitor.csproj:4]: these comments must be addressed by the new owner *before* actual release (the original contributor is no longer working on this so @reyang and myself will own this).

**Pull request merged**